rclcpp/include/rclcpp/publisher_options.hpp

```diff
@@ -72,9 +77,16 @@ struct PublisherOptionsWithAllocator : public PublisherOptionsBase
     auto message_alloc = std::make_shared<MessageAllocatorT>(*this->get_allocator().get());
     result.allocator = rclcpp::allocator::get_rcl_allocator<MessageT>(*message_alloc);
     result.qos = qos.get_rmw_qos_profile();
+
+    // Apply payload to rcl_publisher_options if necessary.
+    if (rmw_implementation_payload && rmw_implementation_payload->has_been_customized()) {
+      rmw_implementation_payload->modify_rcl_publisher_options(result);
```

Comment:
I think you have to update this after https://github.com/ros2/rclcpp/pull/882/commits/ac90f9abf69af2b16c230b8825560a7cefef3ae3.

Reply:
Yes, thanks I missed these. 3cb431c